File: RotateList/RotateList.py
from Solution import *
from crio.ds.List.ListNode import *


def print_list(head):
    while head != None:
        print(head.val, end = " ")
        head = head.next

def main():
   n = int(input().strip())
   lst = None
   lst = input().strip().split()
   lst = [int(x) for x in lst]
   lst = createList(lst)
   k = int(input().strip())
   
   print_list(rotate_list_by_k(lst, k))

if __name__ == '__main__':
    main()

File: RotateList/RotateList.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"
#include "../crio/cpp/io/FastIO.hpp"


int main() {


	FastIO();
	int n, k;
	cin >> n;
	vector<int> list(n);
	for ( int i = 0; i < n; i++)
		cin >> list[i];

	cin >> k;
	ListNode * head = ListNode().createList(list);

	head = rotateList(head, k);
	vector<int> resultList = ListNode().extractList(head);
	for ( int it : resultList) {
		cout << it << " ";
	}
}
